jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -810,24 +810,50 @@ YUI.add('juju-env-api', function(Y) {
         this.setConnectedAttr('maasServer', null);
         return;
       }
-      this.modelGet(data => {
-        if (data.err) {
-          console.warn('error calling ModelGet API: ' + data.err);
+      this.cloudInfo(data.cloud, data => {
+        if (!data) {
```

Comment:
Let's make this impossible by providing (err, cloud) to the callback.
So, for instance
```javascript
this.cloudInfo(data.cloud, (err, cloud) => {
  if (err) {
    console.log('cannot retrieve maas server from cloud info: ' + err);
    this.setConnectedAttr('maasServer', null);
    return;
  }
  this.setConnectedAttr('maasServer', cloud.server);
}
```
We could make the provided cloud a js friendly object, so that we don't have to do the .value stuff.